snabble/Android-SDK
# Add address and email to PAYONE form

For various reasons _(see APPS-1296 in JIRA)_, we have to provide the credit card owners address and email during the addition of the card and in an encrypted form on usage. To do so a necessary form field and implementation processing this data has been added. There is no further validation of the input, except there has to be any. The state only has to be choosen if they are provided for the selected country.

### How to test?

* Install this branch with one of the apps making use of the sdk.
* Choose the testing environment from dev settings.
* Add a new credit cards where the provider is PAYONE.
  * Check that the country the device is set to _(device's language settings)_ is the preselected one.
    * If none can be found the use has to choose one manually.
   * Choose a country where a state is mandatory _(e.g. US, IN, ID, JP, ...)_
   * Check that all field that are visible need to provide some data, otherwise an error toast will appear and card cannot be added.
   * Play around with the fields and selectors and try to break it.

### Definition of Done

- ~~Dokumentation gepflegt~~

#### App Tests
- [x] Environements beachtet _(Production/Staging)_
- [x] Unterstützte Sprachen sind getestet
- [x] Light-/Dark-Mode getestet
